src/main/java/net/minecraftforge/common/Tags.java

```diff
@@ -453,14 +480,23 @@ private static void init(){}
          */
         public static final TagKey<Item> TOOLS_FISHING_RODS = tag("tools/fishing_rods");
```

Comment:
I think we really should ditch all the tool tags in favor of tool actions as the tags disallow NBT sensitive implementation of tools (which I do constantly). It would be trivial to implement a tool action ingredient, but that idea was turned down without discussion when I brought it up before. While we are breaking tags might as well discuss again.

Reply:
I think we should keep them, but definitely encourage functional behaviour to be done using tool actions. Tagging tool types is, in my eyes, more important for crafting recipes and slot interactions (like an automatic fisher which accepts any fishing rod, using its stats to affect how many fish are produced) than actually using the tool.

Reply:
You can use a tool action for a crafting recipe or for a slot interaction.

Reply:
Fabric has these tool tags as well. I am not seeing a tool action in the vanilla recipe json
![image](https://github.com/neoforged/NeoForge/assets/40846040/4f0f3f9c-f3a9-482f-9854-f6b449ebaeb6)

If it is a modded thing, then not having these tools defined by tag makes it harder for datapackers outside Neoforge to make use of modded tools if their datapack gets used on Neoforge. They could have a chest that uses a tag entry to randomly generate any modded or vanilla fishing rod. I too think we we should keep these tool tags for both parity with fabric and help out datapackers.

But, I will definitely add javadoc so that neoforge modders know to use tool actions instead for determining a tool's behavior.

Reply:
Yes, my original comment said the javadocs is pretty terrible. It just states it's "not the same" but states nothing about the difference

Reply:
Ah then yeah this new javadoc should help get neo modders to use the ToolActions rather than the tag for determining behaviors
![image](https://github.com/neoforged/NeoForge/assets/40846040/714b52c9-9b46-4035-9b95-4f17c504a684)